tests/integration/kubernetes/confidential_kbs.sh

```diff
@@ -0,0 +1,354 @@
+#!/usr/bin/env bash
+
+# Copyright (c) 2024 Red Hat
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Provides a library to deal with the CoCo KBS
+#
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+kubernetes_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+# shellcheck disable=1091
+source "${kubernetes_dir}/../../gha-run-k8s-common.sh"
+
+# Where the kbs sources will be cloned
+readonly COCO_KBS_DIR="/tmp/kbs"
+# The k8s namespace where the kbs service is deployed
+readonly KBS_NS="coco-tenant"
+# The kbs service name
+readonly KBS_SVC_NAME="kbs"
+
+# Set "allow all" policy.
+#
+kbs_set_allow_all_policy() {
+	kbs_set_policy "$COCO_KBS_DIR/kbs/sample_policies/allow_all.rego"
+}
+
+# Set policy.
+#
+# Parameters:
+#	$1 - path to policy file
+#
+kbs_set_policy() {
+	local file="${1:-}"
+
+	if [ ! -f "$file" ]; then
+		echo "ERROR: policy file '$file' does not exit"
+		return 1
+	fi
+
+	kbs_install_cli
+
+	kbs-client --url "$(kbs_k8s_svc_http_addr)" config \
+		--auth-private-key "$(kbs_get_private_key)" set-resource-policy \
+		--policy-file "$file"
+}
+
+# Set resource data.
+#
+# Parameters:
+#	$1 - repository name (optional)
+#	$2 - resource type (mandatory)
+#	$3 - tag (mandatory)
```

Comment:
Any particular reason to separate these into three fields? We often just keep them together as the `resource_uri`. Maybe it is clearer with the fields separate idk. Also where do you get the impression that the `repository` field is optional? I'm not sure that's by design.

Reply:
Probably because we often use `default` as repository name.

btw, a `resource_uri` look like
```
kbs:///repository-name/resource-type/tag
```

This hints that the logic should strip the prefix `kbs:///`

Reply:
@fitzthum my reasoning is that splitting in parameters it won't require to document the path format, it keeps hidden that detail from users.

I'm not sure I understood your comment @Xynnn007 . Do I need to prefix the path string with `kbs:///`?

Reply:
ah, ok, thanks @Xynnn007 !